dashboard/src/api/api.js

```diff
@@ -0,0 +1,49 @@
+class Api {
+    token = '';
+
+    async decodeResults(result) {
+        const decoded = await result.json();
+        if (result.status === 401) {
+            throw Error(decoded.error || "Unauthorized")
+        }
+        if (result.status !== 200) {
+            throw Error(`Unexpected status ${result.status}`);
+        }
+        return decoded;
+    }
+    
+    // apiGet performs a GET request on the API with given local URL.
+    // The result is decoded from JSON and returned.
+    async get(localURL) {
+        let headers = {
+            'Accept': 'application/json'
+        };
+        if (this.token) {
+            headers['Authorization'] = `bearer ${this.token}`; 
+        }
+        const result = await fetch(localURL, {headers});
```

Comment:
Just an FYI because fetch can be a bit too low-level sometimes and have cross-browser surprises: https://yarnpkg.com/en/package/axios